### ISSUE TITLE
Interactive window input history navigation with up / down arrows conflicts with Vim extension keybindings

### ISSUE THREAD

**Issue by yopparai** [microsoft/vscode#130425]
Issue Type: <b>Bug</b>

---

I cannot use `UP`/`DOWN` arrow keys for getting previous inputs from interactive window input field.
But I could do it in previous extension versions.

(I talk about this input field)
![image](https://user-images.githubusercontent.com/55656303/128598099-ab6a0c22-772d-4cf9-9f18-ad6b52248802.png)

---

Extension version: 2021.8.1195043623
VS Code version: Code 1.59.0 (379476f0e13988d90fab105c5c19e7abc8b1dea8, 2021-08-04T23:13:20.182Z)
OS version: Linux x64 5.8.0-63-generic snap
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 5 4600H with Radeon Graphics (12 x 1531)|
|GPU Status|2d_canvas: unavailable_software<br>gpu_compositing: disabled_software<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: disabled_off<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: unavailable_software<br>webgl2: unavailable_software|
|Load (avg)|1, 1, 1|
|Memory (System)|15.08GB (6.38GB free)|
|Process Argv|--no-sandbox --force-user-env --unity-launch --crash-reporter-id 417bce1b-d3f2-43a2-bf1b-e2eb11a9b3f6|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|x11|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspyt653:30270858
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyt639:30300192
pythontb:30283811
pythonvspyt551cf:30345471
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
pythondataviewer:30285071
pythonvsuse255:30340121
vscod805:30301674
pythonvspyt200:30340761
vscextlang:30333561
binariesv615:30325510
vsccppwt:30329788
pythonvssor306:30344512
bridge0708:30335490
vstre464cf:30346473

```

</details>

**Comment by greazer** [comment]
@zwangz, we are actively investigating how to best handle this. If I may ask though, what would you prefer, ultimately? Are you suggesting that you prefer that the input box in the interactive window remains VIM disabled? Or would you prefer that VIM mode works there, just make sure that getting history works as well?

**Comment by yopparai** [comment]
@greazer 
I'd prefer enable VIM mode inside interactive window input. Just because it feels more intuitive to have VIM enabled in all input fields. Otherwise, this is not critical at all.

If it delays the release of the history fix, then it's better to remove VIM from this input field I think.

**Comment by zwangz** [comment]
Hi @greazer, thanks for asking. Personally I would love to see vim mode work in the interactive window (loved it when first trying it so well done!) - so yes it'd be great if we can just get history working as well.

**Comment by joyceerhl** [comment]
@yopparai @zwangz Is it accurate to say that you would expect the interactive window's up/down (history navigation) to win over the Vim extension's up/down keybindings? Are there any others where you would expect the interactive window to trump the Vim extension (e.g. esc)?

**Comment by zwangz** [comment]
@joyceerhl Ideally I would imagine the most intuitive way is to use up/down for history nav at least in Vim's insert mode. Or just "ctrl + up/down" like in iPython if that makes it easier? 

Can't think of others where interactive window should override at the moment as frankly I've not used it much yet.  Re "esc" I'd prefer it to behave like in vim unless overridden explicitly. I guess also the less overriding the more consistent it would be with the main editor which is a good thing.

**Comment by yopparai** [comment]
@joyceerhl 
> Is it accurate to say that you would expect the interactive window's up/down (history navigation) to win over the Vim extension's up/down keybindings?

Yes. Also I like @zwangz's option.

> Are there any others where you would expect the interactive window to trump the Vim extension (e.g. esc)?

No. I did not find any other conflicting keybindings.

**Comment by rebornix** [comment]
@zwangz @yopparai thank you both for the suggestions. It makes sense to support history navigation in insert mode.

**Comment by zwangz** [comment]
Can confirm this is fixed in Vim extension version  v1.21.6.

Great work thanks! @rebornix @greazer @joyceerhl
